### ISSUE TITLE
Remove the necessity of ActionPrevious

### ISSUE THREAD

**Issue by Mitmischer** [kivy/kivy#2046]
Hello,

I spent a good deal of time fighting with that issue, that's why I would suggest some clarification here: http://kivy.org/docs/api-kivy.uix.actionbar.html .

```
FloatLayout:
    ActionBar:
```

works. However:

```
FloatLayout:
   ActionBar:
       ActionView:
```

will throw a very confusing exception (see the gist here: https://gist.github.com/Mitmischer/9952538 ) . It only works, if the code segment in quesion looks like that:

```
FloatLayout:
   ActionBar:
       ActionView:
           ActionPrevious:
```

Sadly, in the documentation there is only a succint hint, namely "An ActionView will contain an ActionPrevious, [...]". I expected the ActionView to, well, contain an ActionPrevious ;) - I did not feel responsible for adding one. Therefore I would rephrase that to "An ActionView _must_ contain an ActionPrevious" or make the exception more meaningful. However, if possible I would prefer to have that limitation removed overall. Without the ActionPrevious, ActionBar would serve well as a general-purpose status bar (as there is not always the need for a back-button).
##

**Comment by dessant** [comment]
We are closing some of the stale issues, please create a new issue/pr if you'd like to work on this feature, or get in touch with us on #kivy-dev.

**Comment by S0AndS0** [comment]
While this issue maybe closed it was never resolved so what follows is my attempt at working around the requirements as well as expanding upon where other visible widgets end up.


    #!/usr/bin/env python

    from kivy.app import App
    from kivy.lang import Builder
    from kivy.uix.gridlayout import GridLayout

    ## Un-comment if you wish to add to the action view via Python
    # from kivy.uix.actionbar import ActionButton, ActionGroup

    kv = """
    #:import Window kivy.core.window.Window

    <SomeLayout_GridLayout>:
        cols: 1
        rows: 2
        row_force_default: True
        rows_minimum: {0: ActionBar.height, 1: self.height - ActionBar.height}
        SomeMenu_ActionBar:
            id: ActionBar

    <SomeMenu_ActionBar@ActionBar>:
        ActionView:
            id: ActionView
            ## Choose one of the following
            # HiddenIcon_ActionPrevious: ## Hide just the icon, but keep the text
            # HiddenText_ActionPrevious: ## Keep the icon and UI methods but hide the text
            # Hidden_ActionPrevious: ## Hide both and other widgets go to the right
            Hide_ActionPrevious: ## Hide both and other widgets go to the left

            ## Do the ActionGroup(s) with ActionButton(s) within thing
            ActionGroup:
                id: App_ActionGroup
                mode: 'spinner'
                text: 'App'
                ActionButton:
                    text: 'Settings'
                    on_press: app.open_settings()
                ActionButton:
                    text: 'Quit'
                    on_press: app.get_running_app().stop()

            ActionGroup:
                id: File_ActionGroup
                mode: 'spinner'
                text: 'File'
                ActionButton:
                    text: 'Open'
                ActionButton:
                    text: 'Save'

    ## Inspired by: https://stackoverflow.com/a/36201399/2632107
    ## Hide just the icon, but keep the text, note though
    ##  that one will lose the 'on_press' and similar methods
    <HiddenIcon_ActionPrevious@ActionPrevious>:
        title: app.title if app.title is not None else 'Action Previous'
        with_previous: False
        app_icon: ''
        app_icon_width: 0
        app_icon_height: 0
        size_hint_x: None
        width: len(self.title) * 10

    ## Keep the icon and UI methods but hide the text
    <HiddenText_ActionPrevious@ActionPrevious>:
        with_previous: False
        on_press: print(self)
        title: ''

    ## Hide both but keep ActionGroup(s) and or ActionButton(s) to the right
    <Hide_ActionPrevious@ActionPrevious>:
        with_previous: False
        on_press: print(self)
        title: ''
        app_icon: ''
        app_icon_width: 0
        app_icon_height: 0

    ## Hide everything and allow ActionGroup(s) and or ActionButton(s) to pull to the left
    <Hidden_ActionPrevious@ActionPrevious>:
        with_previous: False
        on_press: print(self) ## method that will not be called easily
        title: '' ## Try placing text here, only a few pixels should show
        size_hint: None, None
        size: 0, 0
    """


    class SomeLayout_GridLayout(GridLayout):
        pass


    class SomeApp(App):
        def build(self):
            ## Cannot set this in '__init__' for some reason
            self.title = 'Some Sweet App'

            Builder.load_string(kv)

            some_layout = SomeLayout_GridLayout()
            ## Uncomment next line if ya wish to use 'add_widget'
            ##  method on ActionView and add ActionGroup(s) and/or
            ##  ActionButton(s) via Python
            # some_actionview = some_layout.ids.ActionBar.ids.ActionView
            return some_layout

    if __name__ == '__main__':
        SomeApp().run()


To the devs of Kivy, feel free to keep this issue closed as this is just a work around for those that have similar desires to be without `ActionPrevious` within their `ActionView`s

**Comment by apacelus** [comment]
@S0AndS0 thank you for this solution
